tests/pyspark/test_pyspark_decorators.py

```diff
@@ -0,0 +1,113 @@
+)
+from tests.pyspark.conftest import spark_df
+
+
+class TestPanderaDecorators:
+    """Class to test all the different configs types"""
+
+    sample_data = [("Bread", 9), ("Cutter", 15)]
+
+    def test_pyspark_cache_requirements(self, spark, sample_spark_schema):
+        """Validates if decorator can only be applied in a proper function."""
+        # Set expected properties in Config object
+        CONFIG.pyspark_cache = True
+        input_df = spark_df(spark, self.sample_data, sample_spark_schema)
+
+        class FakeDataFrameSchemaBackend:
+            """Class that simulates DataFrameSchemaBackend class."""
+
+            @cache_check_obj()
+            def func_w_check_obj(self, check_obj: DataFrame):
+                """Right function to use this decorator."""
+                return check_obj.columns
+
+            @cache_check_obj()
+            def func_wo_check_obj(self, message: str):
+                """Wrong function to use this decorator."""
+                return message
+
+        # Check for a function that does have a `check_obj`
+        with does_not_raise():
+            instance = FakeDataFrameSchemaBackend()
+            _ = instance.func_w_check_obj(check_obj=input_df)
+
+        # Check for a wrong function, that does not have a `check_obj`
+        with pytest.raises(KeyError):
+            instance = FakeDataFrameSchemaBackend()
+            _ = instance.func_wo_check_obj("wrong")
+
+    @pytest.mark.parametrize(
+        "cache_enabled,unpersist_enabled,"
+        "expected_caching_message,expected_unpersisting_message",
+        [
+            (True, True, True, True),
+            (True, False, True, None),
+            (False, True, None, None),
+            (False, False, None, None),
+        ],
+        scope="function",
+    )
+
+    # pylint:disable=too-many-locals
+    def test_pyspark_cache_settings(
+        self,
+        spark,
+        sample_spark_schema,
+        cache_enabled,
+        unpersist_enabled,
+        expected_caching_message,
+        expected_unpersisting_message,
+        caplog,
+    ):
+        """This function validates that caching/unpersisting works as expected."""
+        # Set expected properties in Config object
+        CONFIG.pyspark_cache = cache_enabled
+        CONFIG.pyspark_unpersist = unpersist_enabled
+
+        # Prepare test data
+        input_df = spark_df(spark, self.sample_data, sample_spark_schema)
+        pandera_schema = DataFrameSchema(
+            {
+                "product": Column(T.StringType(), Check.str_startswith("B")),
+                "price_val": Column(T.IntegerType()),
+            }
+        )
+
+        # Capture log message
+        with caplog.at_level(logging.DEBUG, logger="pandera"):
+            df_out = pandera_schema.validate(input_df)
+
+        # Assertions
+        assert isinstance(df_out, DataFrame)
+        if expected_caching_message:
+            assert (
+                "Caching dataframe..." in caplog.text
```

Comment:
Are these logging statements consistent over differnt PySpark versions for which pandera is expected to be compatible with?

Reply:
Otherwise great test!

Reply:
About the logging statements, this statement does not depend on `pyspark` as it is at "python level", using only `logging` module.
I'm setting this debugging log entry in `pandera/backends/pyspark/decorators.py#169`